sympy/core/sympify.py

```diff
@@ -276,6 +276,10 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     if strict:
         raise SympifyError(a)
 
+    if hasattr(a, 'shape'):
```

Comment:
I fear that the `shape` attribute does not identify _numpy_ classes unambiguously. Maybe there are libraries other than _numpy_ defining objects with _shape_ that may be sympified.